build-tools/packages/build-cli/src/commands/merge/branch.ts

```diff
@@ -0,0 +1,214 @@
+        > - Resolve any merge conflicts between this branch and next (and push the resolution to this branch). Merge next into this branch if needed. **Do NOT rebase or squash this branch: its history must be preserved**.
+        > - Ensure CI is passing for this PR, fixing any issues. Please don't look into resolving **Real service e2e test** and **Stress test** failures as they are **non-required** CI failures.
+        For more information about how to resolve merge conflicts and CI failures, visit [this wiki page](https://github.com/microsoft/FluidFramework/wiki/Main-next-Automation).`;
+    const octokit = new Octokit({ auth });
+    const newPr = await octokit.request("POST /repos/{owner}/{repo}/pulls", {
+        owner,
+        repo,
+        title: "Automate: Main Next Integrate",
+        body: description,
+        head: source,
+        base: target,
+    });
+
+    await octokit.request("POST /repos/{owner}/{repo}/issues/{issue_number}/assignees", {
+        owner,
+        repo,
+        // eslint-disable-next-line camelcase
+        issue_number: newPr.data.number,
+        assignees: [author],
+    });
+
+    await octokit.request("POST /repos/{owner}/{repo}/pulls/{pull_number}/requested_reviewers", {
+        owner,
+        repo,
+        // eslint-disable-next-line camelcase
+        pull_number: newPr.data.number,
+        reviewer: [],
+    });
+
+    await octokit.request("POST /repos/{owner}/{repo}/issues/{issue_number}/labels", {
+        owner,
+        repo,
+        // eslint-disable-next-line camelcase
+        issue_number: newPr.data.number,
+        labels: ["main-next-integrate", "do-not-squash-merge"],
+    });
+
+    return newPr.data.number;
+}
+
+export default class MergeBranch extends BaseCommand<typeof MergeBranch.flags> {
+    static description = "Sync branches depending on the batch size passed";
+
+    static examples = ["<%= config.bin %> <%= command.id %>"];
+
+    static flags = {
+        auth: Flags.string({
+            description: "GitHub authentication token",
+            char: "a",
+            required: true,
+        }),
+        source: Flags.string({
+            description: "Source branch",
+            char: "s",
+            required: true,
+        }),
+        target: Flags.string({
+            description: "Target branch",
+            char: "t",
+            required: true,
+        }),
+        batchSize: Flags.integer({
+            description: "Maximum number of commit to include in the pull request",
+            char: "b",
+            required: true,
+        }),
+        pullRequestInfo: Flags.string({
+            description: "Pull request data",
+            char: "p",
+            multiple: true,
+        }),
+        ...BaseCommand.flags,
+    };
+
+    public async run(): Promise<void> {
+        const flags = this.processedFlags;
+
+        const resolvedRoot = await getResolvedFluidRoot();
+        const gitRepo = new GitRepo(resolvedRoot);
```

Comment:
I suggest using the context instead:

```suggestion
        const context = await this.getContext();
        const gitRepo = context.gitRepo;
        const resolvedRoot = gitRepo.resolvedRoot;
```